source_http_test.go

```diff
@@ -124,6 +124,30 @@ func TestHttpImageSourceForwardAuthHeader(t *testing.T) {
 	}
 }
 
+func TestHttpImageSourceCustomHeaders(t *testing.T) {
```

Comment:
Great start! However please include test cases for:
- Proof that the client can't override headers set by Imaginary
- Proof that empty HTTP headers also work
- Proof that case sensitivity when setting headers isn't a problem (headers set by the client are correctly set in the upstream request)
- Proof that custom headers that aren't white-listed, but who are defined by the client, aren't forwarded.